### ISSUE TITLE
Crash when launching 78.0.1 update

### ISSUE THREAD

**Issue by chknatwork** [Tobi823/ffupdater#432]
**Describe the bug**
Crash upon launch on 78.0.1

**To Reproduce**
1. Launch FFUpdatwe

**Expected behavior**
Launch and update browsers when available

**Error message from FFUpdater**
Stacktrace:
```
java.lang.ClassCastException
	at androidx.appcompat.app.ToolbarActionBar$$ExternalSyntheticThrowCCEIfNotNull0.m(SourceFile:0)
	at de.marmaro.krt.ffupdater.network.fdroid.CustomRepositoryConsumer.getLatestUpdate(Unknown Source:112)
	at de.marmaro.krt.ffupdater.network.fdroid.CustomRepositoryConsumer$getLatestUpdate$1.invokeSuspend(Unknown Source:16)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:11)
	at kotlinx.coroutines.DispatchedTask.run(Unknown Source:88)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7918)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@84e5ebb, Dispatchers.Main]
Timestamp: 2023-07-05T17:52:43.731

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 78.0.1 (143) release |
| Device | HD1911 (OnePlus7TPro, OnePlus7TPro, msmnile) |
| Manufacturer | OnePlus (OnePlus) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 13 (SDK: 33) |
| OS | b6a8bf7830ce, root, release-keys, 1688451478000 |

**Comment by TPS** [comment]
I've been getting this error since v78.0.0.  Does it help?
![Pngpro_out_fb3b98a08df9c777f7423d3702d721e5.png](https://github.com/Tobi823/ffupdater/assets/185902/a3e395ce-2364-4911-89a4-acaeabb05e33)

**Comment by Tobi823** [comment]
But you can dismiss the dialog?

**Comment by TPS** [comment]
When I dismiss, the app force closes w/o notice or chance to send a bug report.  When I reopen, everything's (most of the time) fine, but will randomly do the above randomly.

**Comment by Tobi823** [comment]
I don't understand why Proguard (tool to compress APK file and remove unnecessary code) causes these crashes. I will simply disable it.

@TPS Can you wait 5 min and test my new version?

**Comment by Tobi823** [comment]
@TPS Here is the new version with disabled Proguard [ffupdater-release.zip](https://github.com/Tobi823/ffupdater/files/11962561/ffupdater-release.zip)

**Comment by TPS** [comment]
> I don't understand why Proguard (tool to … remove unnecessary code) causes these crashes.

I've read about cases where ProGuard incorrectly removes needed code, & that it has a mechanism to protect needed code.

> @TPS Here is the new version with disabled Proguard [ffupdater-release.zip](https://github.com/Tobi823/ffupdater/files/11962561/ffupdater-release.zip)
> 
> Does it work now?

It seems ok, but the crash never happened regularly — maybe 10% of the time.  I'll check over the next day(s) several times & report back if I run into it again.

**Comment by Tobi823** [comment]
FYI: I got a StrictMode violation in my emulator which was caused by double closing a network connection. Maybe this is the StrictMode violation you sometimes get.

**Comment by TPS** [comment]
> @TPS Here is the new version with disabled Proguard [ffupdater-release.zip](https://github.com/Tobi823/ffupdater/files/11962561/ffupdater-release.zip)

@Tobi823 I, too, just got the "Strict Mode Violation" popup again w/ the above build.

**Comment by Tobi823** [comment]
I found some bugs which cause this warning popup. Should now be fixed in version 78.0.2.
